packages/flutter/lib/src/gestures/drag_details.dart

```diff
@@ -66,6 +74,10 @@ class DragStartDetails {
   /// Defaults to the origin if not specified in the constructor.
   final Offset globalPosition;
 
+  /// The buttons pressed when the pointer contacted the screen (changing buttons
+  /// during a drag cancels the gesture.)
+  final int buttons;
```

Comment:
Having this be "buttons" (plural) and canceling when they change seems optimistic. The precision the user would have to have to press two buttons simultaneously is far too much to be reliably achievable. IMHO we are saying that we only support one-button drags here, and if that's the case, we should just say that and enforce it.

Reply:
Thanks.
Though due to the new API design, it's no longer necessary to include buttons in details, therefore all `buttons` have been removed from detail classes.